src/Vehicle/Vehicle.cc

```diff
@@ -1025,6 +1032,27 @@ void Vehicle::_handleGpsRawInt(mavlink_message_t& message)
     }
 }
 
+void Vehicle::_handleGps2Raw(mavlink_message_t& message)
```

Comment:
@LorenzMeier This doesn't seem correct to me. Isn't the GPS position fused into a best position from multiple GPS in the main GPS mavlink telemetry and then the coordinate should only be update only from that telemetry? This would seem to cause the vehicle coordinate to ping back and forth between the telemetry from multiple GPS.

Reply:
Your concern makes sense. I just assumed both GPS would be reporting the same coordinate, but you are right. Although I am not sure what would be the best way to handle it.

I could be wrong but from my understanding, the "best position fused" you are talking about must be the global_position_int message, which we are taking into consideration as well for updating the _coordinate from vehicle. So maybe we should leave gps_raw_int and gps2_raw only for the GPS fact groups, and leave the _coordinate update to global_position_int? 

I think this would make sense for Ardupilot firmware, but I am not sure how PX4, or how the "standard" mavlink rules apply to this.

Reply:
The main fused lat/lon values come from global_position_int. QGC remembers if it has gotten a global_position_int message. Only in the case where QGC is not getting a global_position_int message will gps_raw_int values be used to update lat/lon for the vehicle as a fallback. I think that is all that is really needed and in reality it's unclear whether the gps_raw_int fallback should be in there at all or whether QGC should only ever rely on global_position_int. But the code has been that way for ages now, so who knows. All of that said it still points to not updating lat/lon from gps2_raw and only updating fact group value from that. So can you remove that code?

@LorenzMeier Any comments on the whole gps_raw_int fallback for lat/lon? Seems possibly not needed.

Reply:
Done! https://github.com/mavlink/qgroundcontrol/pull/9453